chevah/compat
# [#5509] Allow cryptography 3.2 with OpenSSL 1.0.2.

Scope
=====

Allow cryptography 3.2 on platforms with OpenSSL 1.0.2.

This is an issue on `python-package`, where the necessary `CRYPTOGRAPHY_ALLOW_OPENSSL_102` env var is set for normal `compat` tests, but not propagated for elevated tests run through `sudo`.


Changes
=======

Export the relevant env var unconditionally.

**Drive-by change**:
  * updated `pip` to latest version.
  * small fix for AIX
  * support for OpenBSD 6.8 and FreeBSD 12.


How to try and test the changes
===============================

reviewers: @adiroiban 

Please review changes.

Check that `compat` tests ran from this branch are green with `cryptography` 3.2.1 on Amazon 2 and CentOS 7 (check the patch diff for the branch toggle):

https://chevah.com/buildbot/builders/python-package-amzn-2/builds/95
https://chevah.com/buildbot/builders/python-package-rhel-7/builds/166
